File: Leetcode_POTD_2025/7_JULY/1695_Maximum_Erasure_Value.py
class Solution(object):
    def maximumUniqueSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        stellar_seen = set()
        max_lumina = 0
        current_nebula_sum = 0
        voyager_left = 0

        for voyager_right in range(len(nums)):
            while nums[voyager_right] in stellar_seen:
                current_nebula_sum -= nums[voyager_left]
                stellar_seen.remove(nums[voyager_left])
                voyager_left += 1
            stellar_seen.add(nums[voyager_right])
            current_nebula_sum += nums[voyager_right]
            max_lumina = max(max_lumina, current_nebula_sum)
        return max_lumina

File: Leetcode_POTD_2025/7_JULY/1695_Maximum_Erasure_Value.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set<Integer> celestial_seen = new HashSet<>();
        int max_lumina = 0;
        int current_nebula_sum = 0;
        int voyager_left = 0;

        for (int voyager_right = 0; voyager_right < nums.length; ++voyager_right) {
            while (celestial_seen.contains(nums[voyager_right])) {
                current_nebula_sum -= nums[voyager_left];
                celestial_seen.remove(nums[voyager_left]);
                voyager_left++;
            }
            celestial_seen.add(nums[voyager_right]);
            current_nebula_sum += nums[voyager_right];
            max_lumina = Math.max(max_lumina, current_nebula_sum);
        }
        return max_lumina;
    }
}
